x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_review_algorithms.md

```diff
@@ -0,0 +1,246 @@
+  - [Functional requirements](#functional-requirements)
+- [Scenarios](#scenarios)
+
+  - [Rule field doesn't have an update and has no custom value - `AAA`](#rule-field-doesnt-have-an-update-and-has-no-custom-value---aaa) -[**Scenario: `AAA` - Rule field is any type**](#scenario-aaa---rule-field-is-any-type)
+  - [Rule field doesn't have an update but has a custom value - `ABA`](#rule-field-doesnt-have-an-update-but-has-a-custom-value---aba) -[**Scenario: `ABA` - Rule field is any type**](#scenario-aba---rule-field-is-any-type)
+  - [Rule field has an update and doesn't have a custom value - `AAB`](#rule-field-has-an-update-and-doesnt-have-a-custom-value---aab)
+    - [**Scenario: `AAB` - Rule field is any type**](#scenario-aab---rule-field-is-any-type)
+  - [Rule field has an update and a custom value that are the same - `ABB`](#rule-field-has-an-update-and-a-custom-value-that-are-the-same---abb)
+    - [**Scenario: `ABB` - Rule field is any type**](#scenario-abb---rule-field-is-any-type)
+  - [Rule field has an update and a custom value that are NOT the same - `ABC`](#rule-field-has-an-update-and-a-custom-value-that-are-not-the-same---abc)
+    - [**Scenario: `ABC` - Rule field is a number or single line string**](#scenario-abc---rule-field-is-a-number-or-single-line-string)
+    - [**Scenario: `ABC` - Rule field is an array of scalar values**](#scenario-abc---rule-field-is-an-array-of-scalar-values)
+  - [Rule field has an update and a custom value that are the same and the rule base version doesn't exist - `-AA`](#rule-field-has-an-update-and-a-custom-value-that-are-the-same-and-the-rule-base-version-doesnt-exist----aa)
+    - [**Scenario: `-AA` - Rule field is any type**](#scenario--aa---rule-field-is-any-type)
+  - [Rule field has an update and a custom value that are NOT the same and the rule base version doesn't exist - `-BC`](#rule-field-has-an-update-and-a-custom-value-that-are-not-the-same-and-the-rule-base-version-doesnt-exist----bc)
+    - [**Scenario: `-BC` - Rule field is a number or single line string**](#scenario--bc---rule-field-is-a-number-or-single-line-string)
+    - [**Scenario: `-BC` - Rule field is an array of scalar values**](#scenario--bc---rule-field-is-an-array-of-scalar-values)
+
+## Useful information
+
+### Tickets
+
+- [Users can customize prebuilt detection rules](https://github.com/elastic/kibana/issues/174168) epic
+- [Implement single-line string diff algorithm](https://github.com/elastic/kibana/issues/180158)
+- [Implement number diff algorithm](https://github.com/elastic/kibana/issues/180160)
+- [Implement array of scalar values diff algorithm](https://github.com/elastic/kibana/issues/180162)
+
+### Terminology
+
+- **Base version**: Also labeled as `base_version`. This is the version of a rule authored by Elastic and unchanged by a user. The algorithms use it as a point of reference to see if the rule is being updated at all in addition to seeing if the user has customized anything themselves
```

Comment:
```suggestion
- **Base version**: Also labeled as `base_version`. This is the version of a rule authored by Elastic as it is installed from the `security_detection_engine` package, with no customizations to any fields by the user.
```
Not sure about `The algorithms use it as a point of reference to check if the rule is being updated at all in addition to seeing if the user has customized anything themselves`. Whether the rule is being updated at all or not depends on a newer `version` of the rule being available, rather than the `base_version`.